arelle/ModelInstanceObject.py

```diff
@@ -513,7 +513,7 @@ def isDuplicateOf(self, other, topLevel=True, deemP0Equal=False, unmatchedFactsS
             if self.modelXbrl == other.modelXbrl and self.parentElement != other.parentElement:
                 return False
             if not (self.context.isEqualTo(other.context,dimensionalAspectModel=False) and
-                    (not self.isNumeric or self.unit.isEqualTo(other.unit))):
+                    (not self.isNumeric or (self.unit is not None and self.unit.isEqualTo(other.unit)))):
```

Comment:
What if both `self.unit` and `other.unit` are `None`?

Reply:
The use of xValid checks makes sense for specific validations where we know certain conditions can’t be met. However, in this case, we’re dealing with a generic function. Since this code is also used by formula, and Arelle doesn’t and shouldn't restrict formula processing to xValid facts, I don’t think we can expect callers to pre-filter.

Additionally, I noticed a similar issue with context usage in the line above; it can also be None.

Reply:
That's reasonable. Can you add a not None check for self.context on the line above so that it behaves the same?